File: token-validation-jaxrs/src/main/java/no/nav/security/token/support/jaxrs/JwtTokenClientRequestFilter.java
package no.nav.security.token.support.jaxrs;

import jakarta.inject.Inject;
import jakarta.ws.rs.client.ClientRequestContext;
import jakarta.ws.rs.client.ClientRequestFilter;
import no.nav.security.token.support.core.JwtTokenConstants;
import no.nav.security.token.support.core.context.TokenValidationContext;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

import static java.util.Collections.singletonList;

public class JwtTokenClientRequestFilter implements ClientRequestFilter {

    private static final Logger LOG = LoggerFactory.getLogger(JwtTokenClientRequestFilter.class);

    @Inject
    public JwtTokenClientRequestFilter() { }

    @Override
    public void filter(ClientRequestContext requestContext) {

        TokenValidationContext context = JaxrsTokenValidationContextHolder.getHolder().getTokenValidationContext();

        if(context.hasValidToken()) {
            LOG.debug("adding tokens to Authorization header");
            StringBuilder headerValue = new StringBuilder();
            context.getIssuers().forEach(issuer -> {
                LOG.debug("adding token for issuer {}", issuer);
                headerValue.append("Bearer ").append(context.getJwtToken(issuer).getTokenAsString());
            });
            requestContext.getHeaders().put(JwtTokenConstants.AUTHORIZATION_HEADER, singletonList(headerValue.toString()));
        } else {
            LOG.debug("no tokens found, nothing added to request");
        }
    }

}

File: token-validation-jaxrs/src/main/kotlin/no/nav/security/token/support/jaxrs/JwtTokenClientRequestFilter.kt
package no.nav.security.token.support.jaxrs

import jakarta.inject.Inject
import jakarta.ws.rs.client.ClientRequestContext
import jakarta.ws.rs.client.ClientRequestFilter
import no.nav.security.token.support.core.JwtTokenConstants.AUTHORIZATION_HEADER
import no.nav.security.token.support.jaxrs.JaxrsTokenValidationContextHolder.getHolder
import org.slf4j.Logger
import org.slf4j.LoggerFactory

class JwtTokenClientRequestFilter @Inject constructor() : ClientRequestFilter {

    override fun filter(requestContext : ClientRequestContext) {
        val context = getHolder().getTokenValidationContext()

        if (context.hasValidToken()) {
            LOG.debug("Adding tokens to Authorization header")
            val headerValue = context.issuers.joinToString(separator = "") {
                LOG.debug("Adding token for issuer $it")
                "Bearer ${context.getJwtToken(it)?.encodedToken}"
            }
            requestContext.headers[AUTHORIZATION_HEADER] = listOf(headerValue)
        } else {
            LOG.debug("No tokens found, nothing added to request")
        }
    }

    companion object {

        private val LOG : Logger = LoggerFactory.getLogger(JwtTokenClientRequestFilter::class.java)
    }
}